src/constraint_functions.jl

```diff
@@ -93,4 +91,400 @@ function EMB.constraints_level_aux(m, n::HydroStorage, 𝒯, 𝒫, modeltype::En
+            (discharge_level(pq, q+1) - discharge_level(pq, q))
+            for q ∈ Q]
+
+    # Range of discharge segments
+    @constraint(m, [t ∈ 𝒯, q ∈ Q],
+        m[:discharge_segment][n, t, q] ≤
+            capacity(n, t) * (discharge_level(pq, q+1).- discharge_level(pq, q))
+    )
+
+    @constraint(m, [t ∈ 𝒯],
+        m[:flow_out][n, t, water_resource(n)] ==
+            sum(m[:discharge_segment][n, t, q] for q ∈ Q)
+    )
+
+    @constraint(m, [t ∈ 𝒯],
+        m[:cap_use][n, t] ==
+            sum(m[:discharge_segment][n, t, q]* η[q] for q ∈ Q)
+    )
+end
+
+"""
+    EMB.constraints_flow_in(m, n::HydroGenerator, 𝒯::TimeStructure, modeltype::EnergyModel)
+    EMB.constraints_flow_in(m, n::HydroPump, 𝒯::TimeStructure, modeltype::EnergyModel)
+
+Method for creating the constraint on the inlet flow of a node `n`.
+
+!!! tip "`HydroGenerator`"
+    The constraints enforce that the water inlet flow is equal to the outlet flow at each
+    operational period `t`, and hence, preserve conservation of mass.
+!!! note "`HydroPump`"
+    The function [`build_pq_constaints`](@ref) is called for creating the constraint on the
+    capacity utilization.
+    The electricity flow to the unit is equal to the capacity utilization
+    The flow of the inlet resources can be constrained through calling the function
+    [`build_constraint`](@ref).
+"""
+function EMB.constraints_flow_in(m, n::HydroGenerator, 𝒯::TimeStructure, modeltype::EnergyModel)
+    @constraint(m, [t ∈ 𝒯],
+        m[:flow_in][n, t, water_resource(n) ] ==
+            m[:flow_out][n, t, water_resource(n)]
+    )
+end
+function EMB.constraints_flow_in(m, n::HydroPump, 𝒯::TimeStructure, modeltype::EnergyModel)
+    build_pq_constaints(m, n, pq_curve(n), 𝒯)
+    @constraint(m, [t ∈ 𝒯],
+        m[:flow_in][n, t, electricity_resource(n)] ==
+            m[:cap_use][n, t]
+    )
+
+    for c ∈ constraint_data(n)
+        build_constraint(m, n, c, 𝒯, resource(c), "flow_in", "gen_penalty")
+    end
+end
+
+"""
+    EMB.constraints_flow_out(m, n::HydroGenerator, 𝒯::TimeStructure, modeltype::EnergyModel)
+    EMB.constraints_flow_out(m, n::HydroPump, 𝒯::TimeStructure, modeltype::EnergyModel)
+
+Method for creating the constraint on the oulet flow of a node `n`.
+
+!!! tip "`HydroGenerator`"
+    - The function [`build_pq_constaints`](@ref) is called for creating the constraint on the
+      capacity utilization.
+    - The electricity flow from the unit is equal to the capacity utilization.
+    - The flow of the inlet resources can be constrained through calling the function
+      [`build_constraint`](@ref).
+!!! note "`HydroPump`"
+    - The constraints enforce that the water outlet flow is equal to the inlet flow at each
+      operational period `t`, and hence, preserve conservation of mass.
+"""
+function EMB.constraints_flow_out(m, n::HydroGenerator, 𝒯::TimeStructure, modeltype::EnergyModel)
+    build_pq_constaints(m, n, pq_curve(n), 𝒯)
```

Comment:
The call of the function `build_pq_constaints` should be moved to `constraints_capacity` as it is related to the capacity.

Reply:
I will implement this.

Reply:
Included in https://github.com/EnergyModelsX/EnergyModelsRenewableProducers.jl/pull/35/commits/2d39ebfd9ea718cc89ee4de166bd27d565647d5c